src/payment/data/sagas.js

```diff
@@ -20,24 +20,32 @@ import { saga as cybersourceSaga } from '../cybersource';
 import { saga as payPalSaga } from '../paypal';
 import { handleErrors, handleMessages } from '../../feedback';
 
+export function* handleBasketResult(basketResult) {
```

Comment:
Consider going through one more pass on DRY-ing further by including message displaying and error handling in here as well?

Reply:
There's some complication around how errors and messages are handled that make this less than straightforward. I'll address in a refactor later.